pepper-apis/dss-core/src/test/java/org/broadinstitute/ddp/model/activity/instance/validation/ComparisonNumericRuleTest.java

```diff
@@ -0,0 +1,141 @@
+package org.broadinstitute.ddp.model.activity.instance.validation;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.broadinstitute.ddp.TxnAwareBaseTest;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.ActivityInstanceDao;
+import org.broadinstitute.ddp.db.dao.AnswerDao;
+import org.broadinstitute.ddp.db.dao.QuestionDao;
+import org.broadinstitute.ddp.db.dto.ActivityInstanceDto;
+import org.broadinstitute.ddp.model.activity.instance.answer.NumericAnswer;
+import org.broadinstitute.ddp.model.activity.instance.question.NumericQuestion;
+import org.broadinstitute.ddp.model.activity.instance.question.Question;
+import org.broadinstitute.ddp.model.activity.types.ComparisonType;
+import org.broadinstitute.ddp.model.activity.types.RuleType;
+import org.broadinstitute.ddp.util.TestDataSetupUtil;
+import org.broadinstitute.ddp.util.TestFormActivity;
+import org.jdbi.v3.core.Handle;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ComparisonNumericRuleTest extends TxnAwareBaseTest {
+    private static TestDataSetupUtil.GeneratedTestData testData;
+    private static Pair<Long, String> questionIdInstanceGuidPair;
+    private static Question unused;
+
+    @BeforeClass
+    public static void setup() {
+        testData = TransactionWrapper.withTxn(TestDataSetupUtil::generateBasicUserTestData);
+        unused = new NumericQuestion("sid", 1L, 2L, false, false, false, null, null, null, List.of(), List.of());
+        questionIdInstanceGuidPair = TransactionWrapper.withTxn(ComparisonNumericRuleTest::prepareTestData);
+    }
+
+    @Test
+    public void testValidateEqual() {
+        var comparisonRule = ComparisonRule.builder()
+                .type(RuleType.COMPARISON)
+                .referenceQuestionId(questionIdInstanceGuidPair.getLeft())
+                .comparisonType(ComparisonType.EQUAL)
+                .build();
+
+        assertTrue(comparisonRule.validate(unused, createAnswer(10L, questionIdInstanceGuidPair.getRight())));
+        assertFalse(comparisonRule.validate(unused, createAnswer(20L, questionIdInstanceGuidPair.getRight())));
```

Comment:
Shouldn't we test each case separately? If we have an "equals" operator, we expect it to return true for equal objects and false for not equal. This will be full coverage. Otherwise, we do two independent tests with only a positive scenario check.

Reply:
You are correct- we do need to check both cases here. Disregard my above comment and let's leave this as-is!